static/js/calculateSceneEdgesLength.js

```diff
@@ -9,43 +9,70 @@ var calculateSceneEdgesLength = function() {
+  is edited, the heading is collected only in the document where this edition
+  was made. As the heading of other user's pad is not collected, we have to
+  force to recalculate it).
+
+2. Edition was made before the beginning of a scene (any scene mark) -
+  if an edition is made on scene marks besides of resetting the cache of the
+  edges of the current scene we reset the cache of the edges of the previous
+  scene as well. We do it because editions on scene marks may change the length
+  of the previous scene as well, see https://trello.com/c/fvdjvPX0/1906
+*/
+calculateSceneEdgesLength.prototype._getLinesToResetCache = function(line) {
+  var firstElementOfChangedScene = this._getFirstElementOfScene(line);
+  var lastElementOfChangedScene = this._getLastElementOfScene(firstElementOfChangedScene);
+  var linesToResetCache = [firstElementOfChangedScene, lastElementOfChangedScene]; // [1]
+
+  var needUpdateFullSceneAndPreviousOne = $(line).hasClass('sceneMark');
+  if (needUpdateFullSceneAndPreviousOne) { // [2]
+    var firstElementOfPreviousScene = this._getFirstElementOfPreviousScene(firstElementOfChangedScene);
+    var lastElementOfPreviousScene = this._getLastElementOfScene(firstElementOfPreviousScene);
+    var linesToResetCache = linesToResetCache.concat(firstElementOfPreviousScene, lastElementOfPreviousScene);
+  }
+
+  return linesToResetCache;
+}
+
 calculateSceneEdgesLength.prototype._cleanElementDimensionCache = function(elements) {
   _.each(elements, function(element) {
     element.children().get(0)._boundingClientRect = null;
   });
 };
 
-calculateSceneEdgesLength.prototype._getLastElementOfScenes = function(linesChanged) {
-  return this._filterLinesBy(this._getfirstElementOfScene, linesChanged)
-}
-
-calculateSceneEdgesLength.prototype._getFirstElementOfScenes = function(linesChanged) {
-  return this._filterLinesBy(this._getLastElementOfScene, linesChanged)
-}
-
-calculateSceneEdgesLength.prototype._filterLinesBy = function(filter, linesChanged) {
-  return _.chain(linesChanged)
-    .map(filter)
-    .compact()
-    .uniq(function() { return this[0].id }) // remove duplicated lines
-    .value();
-};
+calculateSceneEdgesLength.prototype._getFirstElementOfScene = function(line) {
+  // if the edition is on a scene mark the scene affected is below this line
+  var isLineSceneMark = $(line).hasClass('sceneMark');
+  if (isLineSceneMark) {
+    return $(line).nextUntil('.withHeading').addBack().last().next();
+  }
 
-calculateSceneEdgesLength.prototype._getfirstElementOfScene = function(line) {
   return $(line).prevUntil('.sceneMark').addBack().first();
 };
 
 calculateSceneEdgesLength.prototype._getLastElementOfScene = function(line) {
   return $(line).nextUntil('.sceneMark').addBack().last();
 };
 
+calculateSceneEdgesLength.prototype._getFirstElementOfPreviousScene = function(headingLineOfNextScene) {
+  return $(headingLineOfNextScene).prevUntil('.withHeading').last().prev();
+};
```

Comment:
dúvida: o que isso retorna quando `headingLineOfNextScene` for o heading da 1a cena do roteiro?

Reply:
undefined

Reply:
E é um problema isso, ou tudo bem?

Reply:
o único problema eh que pode tentar setar `undefined._boundingClientRect` que n faz nada mas aparece a mensagem no console. Vou remover as linhas undefined para n aparecer essa msg de erro